frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -164,7 +165,9 @@ export const FormFields = ({
           </Box>
         )}
         {form?.responseMode === FormResponseMode.Encrypt &&
-          form?.payments_field.enabled && (
+          form?.payments_channel.channel !== PaymentChannel.Unconnected &&
+          !!form.payments_field.amount_cents &&
+          !!form.payments_field.description && (
```

Comment:
/nit1: the conditional descent on `form?.` being done inconsistently (first 2 checks versus second 2 checks) looks a little weird. In a case like this, I think it's better to have an explicit check for `form` first:

```suggestion
        {form && form.responseMode === FormResponseMode.Encrypt &&
          form.payments_channel.channel !== PaymentChannel.Unconnected &&
          !!form.payments_field.amount_cents &&
          !!form.payments_field.description && (
```

/nit2: this is the third instance where the code contains the 4 checks to determined `isPaymentEnabled`,  it probably warrants making a utility function for it.

Reply:
Yup we should! Currently only available in the `usePublicForm` hook but we should probably shift it somewhere more accessible for everyone.

Reply:
nit1 done in [27dd088](https://github.com/opengovsg/FormSG/pull/6398/commits/27dd0884622a0fa4c7e96d46b74bca5881040283)